### DISCUSSION TITLE
can abseil-cpp support AbslStringify for char/unsigned char?

### DISCUSSION THREAD

**Discussion by chenchuanyin** [abseil/abseil-cpp#1576] [Q&A] [answered]
When I AbslStringify for type char, but the output of absl::HasAbslStringify<char>::value is false, is it by design?

template <typename Sink>
void AbslStringify(Sink &sink, char u)
{
    sink.Append("%d", u);
}

**Comment by derekmauro** [comment, marked as answer]
This is by design because it is potentially error-prone. Sometimes the user may want to print that char, and sometimes the user may want to print the integer value (consider the type alias `uint8_t`). We decided it is better to be explicit about what it should print:

```
char x = 120;
absl::StrCat(absl::string_view(&x, 1));  // "x"
absl::StrCat(static_cast<int>(x));  // "120"
```

**Reply by chenchuanyin** [reply]
Thank you for the quick reply.

**Reply by MBkkt** [reply]
@derekmauro What do you think about adding explicit type for char as strings?
```
absl::StrCat(absl::Hex{...});
absl::StrCat(absl::Chr2Str{...})
```
The main issue with string_view approach, it doesn't work for rvalue (macros value from 3rdParty library for an example)
